### PR TITLE
Add hidewineexports setting

### PR DESCRIPTION
**Pull request by lucianposton**
Add winetricks settings to toggle hiding wine exports feature available in wine staging. See https://github.com/wine-compholio/wine-staging/blob/master/patches/ntdll-Hide_Wine_Exports/0001-ntdll-Add-support-for-hiding-wine-version-informatio.patch

### CODE REVIEW

**Comment by austin987** [comment]
@slackner @DarkPlayer

**Comment by DarkPlayer** [comment]
Might make sense, especially if you want to add a verb for http://warspear-online.com/en/home

**Comment by slackner** [comment]
The patch looks good and works as expected. There is also a checkbox in winecfg though, so a winetricks recipe is not strictly required. @lucianposton: Is there any specific reason you need this recipe?

**Comment by slackner** [comment]
>Might make sense, especially if you want to add a verb for http://warspear-online.com/en/home

Usually games are out of the scope of winetricks.

**Comment by lucianposton** [comment]
> There is also a checkbox in winecfg though, so a winetricks recipe is not strictly required. @lucianposton: Is there any specific reason you need this recipe?

It's one of the winecfg settings I configure for each new prefix I create. Automating it through winetricks just makes that easier.

Besides the app DarkPlayer mentioned, I don't know of any others that specifically use the hidden procedures (except one that just queries and displays various system info). I do it as a precaution to hide wine/linux host info.

**Comment by austin987** [comment]
It's okay in principle.

Is there a related wine bug? If so, w_workaround_bug() should be used (to warn/explain what his happening). If the bug is a WONTFIX, please add (wine-staging) to the verb description.

**Comment by DarkPlayer** [comment]
The bug is https://bugs.winehq.org/show_bug.cgi?id=38656 which was closed as invalid since the application checks this on purpose. You can consider it as WONTFIX.

**Comment by austin987** [comment]
Sure. I'd like to the verb labelled as wine-staging so if someone asks why it doesn't on plain wine I can point to that :).

Please include that bug in a brief comment

**Comment by lucianposton** [comment]
I updated the branch with the two changes, added "wine-staging" to the verb title and added a comment referencing the wine bug.

File: src/winetricks [hunk 1]
```diff
@@ -18735,6 +18735,42 @@ _EOF_
 
 #----------------------------------------------------------------
 
+w_metadata hidewineexports=enable settings \
+    title="Enable hiding wine exports from Windows (wine-staging)"
```
Comment (austin987) [inline src/winetricks]: This should probably be something like 'Windows applications' or just 'applications'

File: src/winetricks [hunk 2]
```diff
@@ -18735,6 +18735,42 @@ _EOF_
...
+w_metadata hidewineexports=disable settings \
+    title="Disable hiding wine exports from Windows (wine-staging)"
```
Comment (austin987) [inline src/winetricks]: Also here.

File: src/winetricks [hunk 3]
```diff
@@ -18735,6 +18735,42 @@ _EOF_
...
+
+load_hidewineexports()
+{
+    # Wine exports some procedures allowing apps to query the wine version and
```
Comment (austin987) [inline src/winetricks]: 'some procedures' > 'some functions'

File: src/winetricks [hunk 4]
```diff
@@ -18735,6 +18735,42 @@ _EOF_
...
+    # information about the host environment. Using these procedures, some
+    # apps will intentionally terminate if they can detect that they are run
```
Comment (austin987) [inline src/winetricks]: 'run' > 'running'

**Comment by austin987** [comment]
Looking better, thanks. I noticed a few other things while reviewing. Once those are fixed I think it should be good to merge.

**Comment by lucianposton** [comment]
I pushed those requested changes.

**Comment by austin987** [comment]
Thanks, merged.

**Pull request merged**